a10networks/a10-octavia
# STACK-1892: Added support newly added columns in vthunders table.

## Description
Added support for newly added protocol, port and write_memory columns in vthunders table through database task.

## Jira Ticket
https://a10networks.atlassian.net/browse/STACK-1892

## Technical Approach
- Getting the values for protocol, port through config file.
- If they are not specified in config, then default values will be taken which are defined in config_options.py.
- write_memory is always set to True.

## Config Changes
If you want to set port and protocol through config file, then add following in config:
Values of port and protocol are user-defined

[hardware_thunder]
    devices = [
                    {
                     "project_id": "d215d1a6f7d44057922bfc6433b1dcbf",
                     "username" : "admin",
                     "password" : "a10",
                     "ip_address" : "10.43.12.122",
                     "device_name" : "rack_1",
                     "partition_name": "PartitionA",
                     "hierarchical_multitenancy": "enable",
                     **"protocol":"tcp",**
                     **"port":"80"**
                    }

## Manual Testing
1. a10-octavia.conf is having below values:
```
[hardware_thunder]
devices = [
                    {
                     "project_id":"4606fa78c21c427a9964fe158b18c395",
                     "ip_address":"10.0.0.15",
                     "username":"admin",
                     "password":"a10",
                     "device_name":"device1",
                     "port":"443",
                     "protocol": "https"
                     }
          ]

```
2. Create a loadbalancer:
![image](https://user-images.githubusercontent.com/58077446/102319946-a5dde780-3fa1-11eb-98e6-2dcb061c89e3.png)

3. Check vthunders table of octavia db:
![image](https://user-images.githubusercontent.com/58077446/102320089-daea3a00-3fa1-11eb-8871-c064c193f2af.png)
